tawazi/_dag/dag.py

```diff
@@ -657,136 +658,133 @@ def run_subgraph(
 
         return exec_nodes, results, profiles
 
-    def __call__(self, *args: P.args, **kwargs: P.kwargs) -> RVDAG:
-        """Execute the DAG scheduler via a similar interface to the function that describes the dependencies.
+    def _describe_subdag(self, *args: P.args, **kwargs: P.kwargs) -> RVDAG:
```

Comment:
I simply took out a big part of the dag description from __call__ into _describe_subdag